### PR TITLE
Queries section for new book

### PR DESCRIPTION
**Pull request by alice-i-cecile**
Ported from #182.

# Status

- [x] Show, don't tell
- [x] Revisit and revise
- [ ] Add section on `get_multiple_mut` and `iter_combinations`
- [ ] Use "query fetch" and "query filter" terminology

### CODE REVIEW

File: content/learn/book/ecs/queries/_index.md [hunk 1]
```diff
@@ -5,3 +5,459 @@ template = "book-section.html"
...
+// This query requests the Entity identifier of all entities that don't have the Player component,
+// so then they can be passed into our Commands system parameter
+// Q is Entity, and F is Without<Player>
+fn despawn_all_non_player_entities(mut commands: Commands, query: Query<Entity, Without<Player>>) {}
+
+// This systems has two Query system parameters
+// For player_query, Q is (&Position, &mut Targeting) and F is With<Player>
+// For target_query, Q is (Entity, &Position, &TargetPriority) and F is (With<Enemy>, Without<Player>)
+fn select_target(
+    player_query: Query<(&Position, &mut Targeting), With<Player>>,
```
Comment (johnnywong1) [inline content/learn/book/ecs/queries/_index.md:92]: Replace `player_query` with `mut player_query`

File: content/learn/book/ecs/queries/_index.md [hunk 2]
```diff
@@ -5,3 +5,459 @@ template = "book-section.html"
...
+    // We need to use mut query, &mut Age, mut age, and .iter_mut() here because we need mutable access
+    for mut age in query.iter_mut() {
+        // age.0 refers to the first (only) field on our tuple type
+        // We could make this more ergonomic by implementing the Add<Age, u64> trait
+        // or the AddAssign<Age> trait on our Age component type
+        age.0 = age.0 + 1;
+    }
+}
+
+fn take_damage(query: Query<(&mut Life, &mut IncomingDamage)>) {
```
Comment (johnnywong1) [inline content/learn/book/ecs/queries/_index.md:137]: replace `query` with `mut query`

File: content/learn/book/ecs/queries/_index.md [hunk 3]
```diff
@@ -5,3 +5,459 @@ template = "book-section.html"
...
+    Peel,
+    Mince,
+    Juice,
+    Deseed,
+}
+
+#[derive(Component)]
+struct NeedsCooking;
+
+fn prepare_fruits(query: Query<(&mut Fruit, Option<&SpecialPreparation>>, Option<&NeedsCooking>)>){
```
Comment (johnnywong1) [inline content/learn/book/ecs/queries/_index.md:331]: replace `query` with `mut query`

File: content/learn/book/ecs/queries/_index.md [hunk 4]
```diff
@@ -5,3 +5,459 @@ template = "book-section.html"
...
+# 
+# #[derive(Component)]
+# struct Aura;
+#
+# #[derive(Component)]
+# struct Creature;
+
+fn defense_aura_system(
+    aura_query: Query<&Transform, With<Aura>>,
+    target_query: Query<(&mut Defense, &Transform), With<Creature>>) {
```
Comment (johnnywong1) [inline content/learn/book/ecs/queries/_index.md:373]: replace `target_query` with `mut target_query`

File: content/learn/book/ecs/queries/_index.md [hunk 5]
```diff
@@ -5,3 +5,459 @@ template = "book-section.html"
...
+
+```rust
+# use bevy::prelude::*;
+# 
+# #[derive(Component)]
+# struct Player;
+
+fn camera_follow_system(
+    player_query: Query<&Transform, With<Player>>,
+    camera_query: Query<&mut Transform, With<Camera>>,
```
Comment (johnnywong1) [inline content/learn/book/ecs/queries/_index.md:401]: replace camera_query with mut camera_query

File: content/learn/book/ecs/queries/_index.md [hunk 6]
```diff
@@ -5,3 +5,459 @@ template = "book-section.html"
...
+
+```rust
+# use bevy::prelude::*;
+# 
+# #[derive(Component)]
+# struct Player;
+
+fn camera_follow_system(
+    player_query: Query<&Transform, With<Player>>,
+    camera_query: Query<&mut Transform, (With<Camera>, Without<Player>)>,
```
Comment (johnnywong1) [inline content/learn/book/ecs/queries/_index.md:421]: here as well

File: content/learn/book/ecs/queries/_index.md [hunk 7]
```diff
@@ -5,3 +5,459 @@ template = "book-section.html"
...
+Let's try this approach:
+
+```rust
+# use bevy::prelude::*;
+# 
+# #[derive(Component)]
+# struct Player;
+
+fn camera_follow_system(
+    queries: ParamSet<(
```
Comment (johnnywong1) [inline content/learn/book/ecs/queries/_index.md:441]: not sure if you need it here

**Review by johnnywong1 — CHANGES_REQUESTED** [review]

File: content/learn/book/ecs/queries/_index.md (hunk 1 above)
Comment (alice-i-cecile) [reply content/learn/book/ecs/queries/_index.md:92]:
```suggestion
    mut player_query: Query<(&Position, &mut Targeting), With<Player>>,
```

File: content/learn/book/ecs/queries/_index.md (hunk 2 above)
Comment (alice-i-cecile) [reply content/learn/book/ecs/queries/_index.md:137]:
```suggestion
fn take_damage(mut query: Query<(&mut Life, &mut IncomingDamage)>) {
```

File: content/learn/book/ecs/queries/_index.md (hunk 3 above)
Comment (alice-i-cecile) [reply content/learn/book/ecs/queries/_index.md:331]:
```suggestion
fn prepare_fruits(mut query: Query<(&mut Fruit, Option<&SpecialPreparation>>, Option<&NeedsCooking>)>){
```

File: content/learn/book/ecs/queries/_index.md (hunk 4 above)
Comment (alice-i-cecile) [reply content/learn/book/ecs/queries/_index.md:373]:
```suggestion
    mut target_query: Query<(&mut Defense, &Transform), With<Creature>>) {
```

File: content/learn/book/ecs/queries/_index.md (hunk 5 above)
Comment (alice-i-cecile) [reply content/learn/book/ecs/queries/_index.md:401]:
```suggestion
    mut camera_query: Query<&mut Transform, With<Camera>>,
```

File: content/learn/book/ecs/queries/_index.md (hunk 6 above)
Comment (alice-i-cecile) [reply content/learn/book/ecs/queries/_index.md:421]:
```suggestion
    mut camera_query: Query<&mut Transform, (With<Camera>, Without<Player>)>,
```

File: content/learn/book/ecs/queries/_index.md (hunk 7 above)
Comment (alice-i-cecile) [reply content/learn/book/ecs/queries/_index.md:441]:
```suggestion
    mut queries: ParamSet<(
```

**Comment by cart** [comment]
Closing as we will very likely start new PRs (and use this as a reference) when we pick up the new book effort again.

**Pull request closed without merging**